cores/esp8266/hardware_reset.h

```diff
@@ -0,0 +1,12 @@
+#ifndef HARDWARE_RESET_H
+#define HARDWARE_RESET_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+[[noreturn]] extern void hardware_reset(void);
```

Comment:
No need for `extern`

---

`__attribute__((noreturn))`?
at the end, once in header https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-noreturn-function-attribute

In case of `__cplusplus`, were are ok and can use `[[noreturn]]`
Without `__cplusplus`, there is no `noreturn` but `_Noreturn` (or, `noreturn` w/ `stdreturn.h`, that disables itself in C++ mode).

But, I am still having warnings when using either one, so not really sure whats the deal here
> warning: 'noreturn' attribute ignored [-Wattributes]
> warning: '_Noreturn' attribute ignored [-Wattributes]

>  7 | [[noreturn]] void testme(void);
      | ^